### PR TITLE
remove support for debian masters in GCE

### PR DESCRIPTION
**Pull request by mikedanese**
Asked about this on the mailing list and no one objects.

@zmerlynn @roberthbailey 

```release-note
Remove support for debian masters in GCE kube-up.
```

### CODE REVIEW

**Comment by roberthbailey** [comment]
Can we print an error message in kube-up.sh that will be seen on the client side rather than the user seeing the cluster api just never become available and having to ssh into the master to see why? Maybe right at the top of https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/util.sh?

**Comment by roberthbailey** [comment]
Will gutting this have any effect on aws kube-up?

"This script is re-used on AWS.  Some of the above functions will be replaced."

**Comment by roberthbailey** [comment]
@justinsb

File: cluster/gce/configure-vm.sh [hunk 1]
```diff
@@ -1129,17 +1034,9 @@ if [[ -z "${is_push}" ]]; then
   set-kube-env
   auto-upgrade
   ensure-local-disks
-  [[ "${KUBERNETES_MASTER}" == "true" ]] && mount-master-pd
```
Comment (roberthbailey) [inline cluster/gce/configure-vm.sh]: you can also rip out the entire `mount-master-pd` function. Also `find-master-pd`.

File: cluster/gce/configure-vm.sh [hunk 2]
```diff
@@ -1066,15 +982,8 @@ function salt-grains() {
 function configure-salt() {
   mkdir -p /etc/salt/minion.d
   salt-run-local
-  if [[ "${KUBERNETES_MASTER}" == "true" ]]; then
-    salt-master-role
```
Comment (roberthbailey) [inline cluster/gce/configure-vm.sh]: The whole `salt-master-role` function can be removed.

**Comment by justinsb** [comment]
Yes, this is reused on AWS.

Here's the logic:

```
  (
    # Include the default functions from the GCE configure-vm script
    sed '/^#+AWS_OVERRIDES_HERE/,$d' "${KUBE_ROOT}/cluster/gce/configure-vm.sh"
    # Include the AWS override functions
    cat "${KUBE_ROOT}/cluster/aws/templates/configure-vm-aws.sh"
    cat "${KUBE_ROOT}/cluster/aws/templates/format-disks.sh"
    # Include the GCE configure-vm directly-executed code
    sed -e '1,/^#+AWS_OVERRIDES_HERE/d' "${KUBE_ROOT}/cluster/gce/configure-vm.sh"
  ) > "${BOOTSTRAP_SCRIPT}"
```

So it's fine to remove e.g. `salt-master-role`, because we override that in `configure-vm-aws.sh`.

But for example, you can't remove `create-salt-master-auth`, nor the function that calls it, without breaking AWS.

I think we have a few options:

1) nuke aws kube-up
2) don't do this PR
3) pare down this PR dramatically to the functions that don't break AWS
4) copy and paste the GCE code back into the AWS code, so that then you can change the GCE code at will again.  Effectively though this is the same as option 1, because it means AWS has no way to keep up with GCE kube-up changes.

My preference is option 1.  We had already planned to do it, it's just a question of 1.6 vs 1.7

**Comment by whitlockjc** [comment]
How good is `kube-up.sh` for AWS right now?  While I realize there are better options, it would be nice to have an equivalent `kube-up.sh` experience for AWS.  If that means porting things between the two, it will suck but it possible.  Wouldn't be a bad idea to figure out how to share common, OS/cloud agnostic stuff like this.  Of course, this assumes that `kube-up.sh` is still being maintained as the official way to spin things up.

**Comment by justinsb** [comment]
> How good is kube-up.sh for AWS right now?

It is almost certainly broken on every platform other than GCE.  I don't know of anyone that intends to fix it on AWS.

**Comment by whitlockjc** [comment]
Ouch...thanks for the heads up.

**Comment by roberthbailey** [comment]
I've tagged this for the 1.6 branch since we don't want to ship a 1.6 release with broken startup scripts.

**Comment by roberthbailey** [comment]
Now that https://github.com/kubernetes/kubernetes/pull/42196 has merged we don't need to worry about supporting aws kube-up.

**Comment by roberthbailey** [comment]
While rebasing, can you remove the comments like "This script is re-used on AWS. Some of the above functions will be replaced."

File: cluster/gce/configure-vm.sh (hunk 2 above)
Comment (mikedanese) [inline cluster/gce/configure-vm.sh]: Done.

**Comment by whitlockjc** [comment]
@roberthbailey Would it make sense to tag 42077 and 42079 for 1.6 as well?  Your reasoning of _"we don't want to ship a 1.6 release with broken startup scripts"_ seems to apply to those as well.  Of course we'd need to agree on the implementation but I'd be happy to update those PRs accordingly.

**Comment by mikedanese** [comment]
@k8s-bot ok to test

**Comment by mikedanese** [comment]
@k8s-bot gci gce e2e test this

**Comment by spxtr** [comment]
@k8s-bot cvm gke e2e test this

**Comment by spxtr** [comment]
@k8s-bot gci gke e2e test this

**Comment by spxtr** [comment]
@k8s-bot cvm gke e2e test this

**Comment by spxtr** [comment]
@k8s-bot gci gke e2e test this

**Comment by mikedanese** [comment]
@k8s-bot kops aws e2e test this

**Comment by justinsb** [comment]
@k8s-bot kops aws e2e test this

**Pull request merged**